File: problems/1277/solution.cpp
class Solution {
public:
    int countSquares(vector<vector<int>>& matrix) {
        if (matrix.size() == 0 || matrix[0].size() == 0) {
            return 0;
        }
        
        int ans = 0;
        
        for (int i = 0; i < matrix.size(); i++) {
            for (int j = 0; j < matrix[0].size(); j++) {
                if (i > 0 && j > 0 && matrix[i][j] > 0) {
                    matrix[i][j] = min(min(matrix[i - 1][j], matrix[i][j - 1]), matrix[i - 1][j - 1]) + 1;
                }
                
                ans += matrix[i][j];
            }
        }
        
        return ans;
    }
};

File: problems/1277/solution.py
class Solution:
    def countSquares(self, matrix: List[List[int]]) -> int:
        if not len(matrix) or not len(matrix[0]):
            return 0
        
        ans = 0
        
        for i in range(len(matrix)):
            for j in range(len(matrix[0])):
                if i and j and matrix[i][j]:
                    matrix[i][j] = min(matrix[i - 1][j], matrix[i][j - 1], matrix[i - 1][j - 1]) + 1
                
                ans += matrix[i][j]
        
        return ans